File: observer/php/subject.php
<?php
/**
 * file subject.php
 *
 * @author kujaomega
 */

/**
 * This interface will be implemented by the observed objects.
 *
 * interface Subject
 */
interface Subject
{
    /**
     * This is the registerObserver method.
     *
     * @param Observer observer
     */
    public function registerObserver( Observer $observer );

    /**
     * This method is the removeObservers.
     *
     * @param Observer observer
     */
    public function removeObserver( Observer $observer );

    /**
     * This is the notifyObservers method.
     */
    public function notifyObservers();
}

File: observer/java/Subject.java
/*
 * File subject.java
 * 
 * @author kujaomega
 */

/**
 * 
 * Interface subject
 */
public interface Subject 
{
    /**
     * This is the registerobserver.
     */
    public void registerObserver( Observer observer );
    
    /**
     * This is the removeobserver.
     */
    public void removeObserver( Observer observer);
    
    /**
     * With this method we notify the observers.
     */
    public void notifyObservers();
}
